lib/Readers/ELFSection.cpp

```diff
@@ -160,13 +161,25 @@ Fragment *ELFSection::getFirstFragmentInRule() const {
   return nullptr;
 }
 
+uint64_t ELFSection::pAddr() const {
+  OutputSectionEntry *OE = getOutputSection();
+  if (!OE)
+    return 0;
+  return OE->pAddr();
+}
+
+void ELFSection::setPaddr(size_t A) {
```

Comment:
It should be an assert to call `setPaddr` on input sections that do not have any corresponding `OutputSectionEntry`. Or perhaps, we should not expose this function through `ELFSection` and the users that want to set physical address should use:

```cpp
auto *OSE = S->getOutputSectionEntry();
ASSERT(OSE != nullptr, ...);
OSE->setPaddr(...);
```